test/bin/manage_hypervisor_config.sh

```diff
@@ -79,11 +92,17 @@ action_cleanup() {
     fi
 
     # Storage pool
-    if sudo virsh pool-info "${VM_STORAGE_POOL}" &>/dev/null; then
-        sudo virsh pool-destroy "${VM_STORAGE_POOL}"
-        sudo virsh pool-undefine "${VM_STORAGE_POOL}"
-    fi
-
+    for scenario in "${SCENARIO_SOURCES}"/*.sh; do
```

Comment:
This should loop over the pools that actually exist by looking for them in the expected location. That way we only clean up what is actually created.